src/main/java/ui/TestController.java

```diff
@@ -79,4 +85,50 @@ public static boolean isCloseOnQuit() {
         return hasUI() && commandLineArgs.getOrDefault("closeonquit", "false").equalsIgnoreCase("true");
     }
 
+    /**
+     * Creates a Preferences that store data in test config file if run with test mode
+     * or in a default config file specified in the Preferences class
+     * @return
+     */
+    public static Preferences createApplicationPreference() {
+        if (isTestMode()) {
+            return createTestPreferences();
+        }
+
+        return new Preferences(Preferences.GLOBAL_CONFIG_FILE);
+    }
+
+    /**
+     * Creates a Preferences that stores data in the test directory
+     * @return
+     */
+    public static Preferences createTestPreferences() {
+        return new Preferences(Preferences.TEST_CONFIG_FILE);
+    }
+
+    /**
+     * Creates a RepoIO for the application that uses different components
+     * depending on various test options: --test, --testjson etc.
+     * @return
+     */
+    public static RepoIO createApplicationRepoIO() {
+        if (isTestMode()) {
+            return new RepoIO(Optional.of(new DummySource()),
+                              isTestJSONEnabled() ? Optional.of(new JSONStoreStub()) : Optional.empty(),
```

Comment:
Yeah, I think we should use an entirely different name along the line of `disable/enable`JSON store. However, the current meaning of `--testjson` has been used in all guitests. We can open another issue for that since it's quite error prone to refactor from my experience and isn't really related to this PR.